### ISSUE TITLE
Inchi vs smiles isomorphism check is pruning molecules without warning

### ISSUE THREAD

**Issue by lilyminium** [mattwthompson/ib#7]
As part of comparing the old Sage scripts with the new IB benchmarking, I'm looking into why ~3340 conformers (QCArchive IDs) out of 73982 are successfully processed with the old scripts but not (consistently) with `ibstore`. 

One thing that popped up is the inchi vs SMILES mismatch. During _minimize_blob, there's an isomorphism check:

https://github.com/mattwthompson/ib/blob/c2bac8aee87c158c73dcdabfe39c84e69e1e0679/ibstore/_minimize.py#L35-L44

1) Unfortunately this prunes certain structures, e.g. the attached record (data parsed and created with the most recent commit i.e. https://github.com/mattwthompson/ib/commit/c2bac8aee87c158c73dcdabfe39c84e69e1e0679 )

To reduce the problem down, `[N+](=[O])-[O-]` gets turned into `[N](=O)=O`:

```python
>>> mol = Molecule.from_mapped_smiles(
    "[H:37][c:15]1[c:16]([c:17]([c:18]([c:19]([c:14]1[C:13]([H:35])([H:36])[N:2]([C:1]([H:23])([H:24])[H:25])[C:3]([H:26])([H:27])[C:4]([H:28])([H:29])[c:5]2[c:6]([c:7]([c:8]([c:10]([c:12]2[H:34])[O:11][H:33])[O:9][H:32])[H:31])[H:30])[H:40])[H:39])[N+:20](=[O:22])[O-:21])[H:38]",
    allow_undefined_stereo=True
)
>>> mol.to_smiles()
'[H]c1c(c(c(c(c1C([H])([H])N(C([H])([H])[H])C([H])([H])C([H])([H])c2c(c(c(c(c2[H])O[H])O[H])[H])[H])[H])[H])[N+](=O)[O-])[H]'
>>> Molecule.from_inchi(mol.to_inchi(fixed_hydrogens=True), allow_undefined_stereo=True).to_smiles()
'[H]c1c(c(c(c(c1C([H])([H])N(C([H])([H])[H])C([H])([H])C([H])([H])c2c(c(c(c(c2[H])O[H])O[H])[H])[H])[H])[H])N(=O)=O)[H]'
```

This is using OpenEye. With RDKit, this chemical structure at least appears to be recapitulated:

```python
from openff.toolkit.utils.toolkits import RDKitToolkitWrapper
from ibstore import MoleculeStore

wrapper = RDKitToolkitWrapper()
store = MoleculeStore("ib-industry_failure-single.sqlite")
inchi_key = store.get_inchi_keys()[0]
Molecule.from_inchi(inchi_key, toolkit_registry=wrapper).to_smiles()
```
```
'[H]c1c(c(c(c(c1C([H])([H])N(C([H])([H])[H])C([H])([H])C([H])([H])c2c(c(c(c(c2[H])O[H])O[H])[H])[H])[H])[H])[N+](=O)[O-])[H]'
```

I am re-running the script now to see if using RDKit fixes most of it, or if RDKit results in other issues.

I'm not sure if I'm asking for any particular fix here beyond the request below, just tracking progress -- the 3340/73982 figure of ~5% matches the number of [failures OpenEye themselves sees in ChEMBL](https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemFunctions/OEMolToInChI.html). Is the check strictly necessary -- could there maybe be a way to turn it off?

2. Would it be possible to log or put in a warning here when conformers are skipped due to non-isomorphism?

**Other**

I've attached a notebook working out the issue, a single-conformer dataset (Optimization dataset and MoleculeStore database), and my environment.

[isomorphism-issue.zip](https://github.com/mattwthompson/ib/files/12851311/isomorphism-issue.zip)

**Comment by j-wags** [comment]
> To reduce the problem down, [N+](=[O])-[O-] gets turned into [N](=O)=O:

We could fix this category of problems (which I guess are summarized as "offmol-to-inchi-to-offmol conversion arbitrarily switches resonance forms") by changing the isomorphism check to only compare element+connectivity+stereochemistry, and setting `bond_order_matching=False` and `formal_charge_matching=False`. Without fully understanding the purpose of this check, I suspect this would be safe, since the InChI records the total charge so any changes in bond order/formal charge that arise during the roundtrip would just be from resonance. And without looking closely, it looks like we're not using any molecules derived from the InChI for parameter assignment (where resonance changes may lead to different parameter assignment) - instead it's using mapped smiles, which are quite stable for OFFMol roundtrips.

**Comment by mattwthompson** [comment]
This is probably a red herring - I don't think anything InChi needs to be in this codebase.

**Comment by mattwthompson** [comment]
Resolved #8 (I hope)
